examples/vision/mixUp.py

```diff
@@ -0,0 +1,219 @@
+"""
+Title: mixup augmentation for image classification
+Author: [Sayak Paul](https://twitter.com/RisingSayak)
+Date created: 2020/03/06
+Last modified: 2020/03/06
+Description: How to implement mixup to augment images for image classification problems.
+"""
+"""
+## Introduction
+"""
+
+"""
+mixup is a *domain-agnostic* data augmentation technique proposed in [mixup: Beyond
+Empirical Risk Minimization](https://arxiv.org/abs/1710.09412) by Zhang et al. It's
+implemented as the following -
+
+* $\tilde{x}=\lambda x_{i}+(1-\lambda) x_{j}$, where $x_{i}$ and $x_{j}$ are input
+features
+* $\bar{y}=\lambda y_{i}+(1-\lambda) y_{j}$, where $y_{i}$ and $y_{j}$ are one-hot
+encoded labels
+
+The technique is quite holistically named - we are literally mixing up the features and
+their corresponding labels. Implementation-wise it's simple. As you can imagine this
+recipe can be extended to a variety of data modalities such as computer vision, natural
+language processing, speech, and so on.
```

Comment:
Suggestion: minor refactoring

```suggestion
The technique is quite holistically named - we are literally mixing up the features and
their corresponding labels. Implementation-wise it's simple. The data augmentation technique
can be extended to a variety of data modalities such as computer vision, natural language
processing, speech, and so on.
```